components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2762,6 +2762,14 @@ public APIStateChangeResponse changeLifeCycleStatus(String orgId, ApiTypeWrapper
                 apiOrApiProductId = apiMgtDAO.getAPIID(uuid);
                 workflowType = WorkflowConstants.WF_TYPE_AM_API_STATE;
             }
+            JSONObject apiLogObject = new JSONObject();
+            apiLogObject.put(APIConstants.AuditLogConstants.NAME, apiName);
+            apiLogObject.put(APIConstants.AuditLogConstants.CONTEXT, apiContext);
+            apiLogObject.put(APIConstants.AuditLogConstants.VERSION, apiVersion);
+            apiLogObject.put(APIConstants.AuditLogConstants.PROVIDER, providerName);
+            APIUtil.logAuditMessage(APIConstants.AuditLogConstants.API, apiLogObject.toString(),
+                    APIConstants.AuditLogConstants.LIFECYCLE_CHANGED, this.username);
```

Comment:
We have to move this just after `LifeCycleUtils.changeLifecycle()` method call.

There can be a case where there are workflows enabled for Change Lifecycle process. In that case, API Lifecycle won't be changed just after doing the `POST /change-lifecycle` call. There will be a workflow getting triggered and an admin has to approve the lifecycle change request. Then only the actual lifecycle will be changed.

https://apim.docs.wso2.com/en/latest/design/advanced-topics/adding-an-api-state-change-workflow/

Reply:
Updated with: https://github.com/wso2/carbon-apimgt/pull/11406/commits/da765bdc146d08a79f88275c37715846a76b548a